### PR TITLE
[snapshot] Add override_status_bar option

### LINKED ISSUE #15319: Support of new StatusBar overrides in iOS 13 when running Snapshot
### Feature Request
Support of new StatusBar overrides in iOS 13
#### Motivation Behind Feature
Because it's the new way to do it

#### Feature Description
`xcrun simctl status_bar <device> override --time "9:41" --batteryState charged --batteryLevel 100`  more info: `xcrun simctl help status_bar`

### PR DESCRIPTION
**Pull request by bouk**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

xcode now has a way to override the status bar built-in, so there's no need for SimulatorStatusMagic any more.

### Description

This will override the status bar with simctl to show 9:41 AM, and have full battery and reception.

Fixes https://github.com/fastlane/fastlane/issues/15319

### CODE REVIEW

**Comment by gelosi** [comment]
Cool MR! What if we can set the date? Instead of hardcoded 9:41? provide parameter with valid date-time string, using well known format? YYYY-MM-DD hh:mm

File: snapshot/lib/snapshot/simulator_launchers/simulator_launcher_base.rb [hunk 1]
```diff
@@ -102,6 +103,26 @@ def add_media(device_types, media_type, paths)
...
+    def override_status_bar(device_type)
+      device_udid = TestCommandGenerator.device_udid(device_type)
+
+      UI.message("Launch Simulator #{device_type}")
+      Helper.backticks("xcrun instruments -w #{device_udid} &> /dev/null")
+
+      UI.message("Overriding Status Bar")
+
+      # The time needs to be passed as ISO8601 so the simulator formats it correctly
+      time = Time.new(2007, 1, 9, 9, 41, 0)
```
Comment (janpio) [inline snapshot/lib/snapshot/simulator_launchers/simulator_launcher_base.rb:115]: Making this configurable can be tackled in a future PR <3

**Review by janpio — APPROVED** [review]
Looks good to me, someone with a Mac handy should test it though :)

**Review by joshdholtz — APPROVED** [review]
This worked perfectly! Thank you so much for adding this ❤

**Pull request merged**